lib/index.js

```diff
@@ -2850,12 +2858,16 @@ addToplevelMethod('pipeline', function (/*through...*/) {
         });
     });
 
+    wrapper.writable = true;
+
     wrapper.write = function (x) {
         return start.write(x);
     };
 
     wrapper.end = function () {
-        return start.end();
+        var result = start.end();
+        wrapper._onEnd();
```

Comment:
I don't think this is correct. It's a little weird, but writing `end` to a pipeline doesn't end the stream. That's because the internal transforms may be responsible for producing more data from the `nil` token. You just need to set `wrapper.writable = false` here.